kubespawner/spawner.py

```diff
@@ -782,13 +797,43 @@ def start(self):
             raise Exception(
                 'Can not create user pod %s already exists & could not be deleted' % self.pod_name)
 
-        while True:
-            pod = self.pod_reflector.pods.get(self.pod_name, None)
-            if pod is not None and self.is_pod_running(pod):
-                break
-            yield gen.sleep(1)
+        # Note: The self.start_timeout here is kinda superfluous, since
+        # there is already a timeout on how long start can run for in
+        # jupyterhub itself.
+        yield exponential_backoff(
+            lambda: self.is_pod_running(self.pod_reflector.pods.get(self.pod_name, None)),
+            'pod/%s did not start in %s seconds!' % (self.pod_name, self.start_timeout),
+            timeout=self.start_timeout
+        )
+
+        pod = self.pod_reflector.pods[self.pod_name]
+        self.pod_reflector.add_deletion_callback(pod.metadata.name, self.stop_notify)
         return (pod.status.pod_ip, self.port)
 
+    def start_polling(self):
```

Comment:
I think you can already set `poll_interval = 0` to disable polling without overriding the methods.